### PR TITLE
Update testing chapter

### PR DESCRIPTION
**Pull request by presidento**
- setup, teardown methods
- use built-in std library notation

### CODE REVIEW

**Comment by presidento** [comment]
Draft: waiting for https://github.com/nushell/nushell/pull/8850

**Comment by presidento** [comment]
It is ready to be merged. (Or after the next release, when std module will be shipped with Nushell.)

File: book/testing.md [hunk 1]
```diff
@@ -1,8 +1,61 @@
-# Testing in Nushell (preview)
+# Testing in Nushell
```
Comment (fdncred) [inline book/testing.md:1]: This is not really "testing in nushell" is it? It looks like it's testing in nushell's standard library, or testing in the nushell language.

So, for me, the filename is misleading and this topic is misleading.

Comment (presidento) [reply book/testing.md:1]: This chapter is how to test your code in Nushell. (I mean, in production environment.) A documentation of the unit testing framework available in Nushell. If you write a code in Nushell, how to test it in Nushell.

Can you suggest a better filename and topic for it? I would be happy to rename it to whatever you want.

Comment (fdncred) [reply book/testing.md:1]: Maybe, "Testing your nushell code"? I'm using "code" to help cover scripts, modules, overlays (at some point), etc.

Comment (presidento) [reply book/testing.md]: I renamed the topic as you suggested. 👍

**Review by amtoine — APPROVED** [review]
looks about right :ok_hand: 

and i like "Testing your nushell code" better as suggested :+1:

File: book/testing.md (hunk 1 above)
Comment (fdncred) [reply book/testing.md]: Thanks

**Pull request merged**